### PR TITLE
222 : Add a flag for Marking PlanDefintion a template

### PR DESCRIPTION
**Pull request by rehammuzzamil**

### CODE REVIEW

File: src/main/java/org/smartregister/domain/PlanDefinition.java [hunk 1]
```diff
@@ -48,6 +48,9 @@
 	@JsonProperty
 	@SerializedName("action")
 	private List<Action> actions;
+
+	@JsonProperty
+	private Boolean isTemplate;
```
Comment (githengi) [inline src/main/java/org/smartregister/domain/PlanDefinition.java:53]: this should be named  `experimental`

Comment (rehammuzzamil) [reply src/main/java/org/smartregister/domain/PlanDefinition.java]: Updated

File: src/main/java/org/smartregister/domain/PlanDefinition.java [hunk 2]
```diff
@@ -48,6 +48,9 @@
 	@JsonProperty
 	@SerializedName("action")
 	private List<Action> actions;
+
+	@JsonProperty
+	private Boolean experimental;
```
Comment (githengi) [inline src/main/java/org/smartregister/domain/PlanDefinition.java:53]: this should be a primitive and maybe false by default

File: src/main/java/org/smartregister/domain/PlanDefinition.java [hunk 3]
```diff
@@ -144,7 +147,15 @@ public Long getServerVersion() {
 	public void setServerVersion(Long serverVersion) {
 		this.serverVersion = serverVersion;
 	}
-	
+
+	public Boolean getExperimental() {
```
Comment (githengi) [inline src/main/java/org/smartregister/domain/PlanDefinition.java:151]: This should be `isExperimental`  since its a boolean

Comment (githengi) [reply src/main/java/org/smartregister/domain/PlanDefinition.java:151]: also should be primitive boolean

File: src/main/java/org/smartregister/domain/PlanDefinition.java (hunk 2 above)
Comment (rehammuzzamil) [reply src/main/java/org/smartregister/domain/PlanDefinition.java]: Changed to primitive and by default it is false.

File: src/main/java/org/smartregister/domain/PlanDefinition.java (hunk 3 above)
Comment (rehammuzzamil) [reply src/main/java/org/smartregister/domain/PlanDefinition.java]: Updated method name.

**Pull request merged**